File: src/main/java/br/com/sd/services/BoardService.java
package br.com.sd.services;

import br.com.sd.entitys.Board;
import br.com.sd.entitys.Pixel;

import java.util.ArrayList;
import java.util.List;

public class BoardService {

    private Board board;

    public Board createBoard(int height, int width) {
        this.board = new Board(height, width);
        return board;
    }

    public Board getBoard() {
        return board;
    }

    public boolean setPixel(Pixel pixel) {
        if (board == null) {
            throw new IllegalStateException("Board não foi criado");
        }

        // valida se o pixel está dentro do quadro
        if (pixel.getX() < 0 || pixel.getX() >= board.getWidth() ||
                pixel.getY() < 0 || pixel.getY() >= board.getHeight()) {
            return false;
        }

        this.board.setPixel(pixel.getX(), pixel.getY(), pixel.getColor());
        return true;
    }

    public String toString() {
        if (board == null) {
            throw new IllegalStateException("Board não foi criado");
        }

        return this.board.toString();
    }
}

File: pyapp/services/boardService.py
from entitys.board import Board
from entitys.pixel import Pixel

class BoardService:
    board: Board

    def __init__(self, height: int, width: int):
        self.board = Board(height, width);
        return self.board;

    def loadBoard(self, board: Board) -> Board:
        self.board = board;
        return board;

    def getBoard(self) -> Board:
        return self.board;

    def setPixel(self, pixel: Pixel):
        if (self.board == None):
            raise AssertionError("Board não foi criado")

        if (pixel.getX() < 0 or pixel.getX() >= self.board.getWidth() or
                pixel.getY() < 0 or pixel.getY() >= self.board.getHeight()):
            return False;
        
        self.board.setPixel(pixel.getX(), pixel.getY(), pixel.getColor());
        return True;

    def toString(self):
        if (self.board == None):
            raise AssertionError("Board não foi criado");

        return self.board.toString();
    
